compilers/concrete-compiler/compiler/include/concretelang/Dialect/FHE/IR/FHEAttrs.td

```diff
@@ -0,0 +1,47 @@
+//===- FHEAttrs.td - FHE dialect attrs ----------------*- tablegen
+//-*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef CONCRETELANG_DIALECT_FHE_IR_FHE_ATTRS
+#define CONCRETELANG_DIALECT_FHE_IR_FHE_ATTRS
+
+include "concretelang/Dialect/FHE/IR/FHEDialect.td"
+include "concretelang/Dialect/FHE/IR/FHETypes.td"
+include "mlir/IR/AttrTypeBase.td"
+include "mlir/IR/OpBase.td"
+
+class FHE_Attr<string name, string attrMnemonic, list<Trait> traits = []>: AttrDef<FHE_Dialect, name, traits> {
+    let mnemonic = attrMnemonic;
+}
+
+def FHE_PartitionAttr: FHE_Attr<"Partition", "partition"> {
+
+    let summary = "An attribute representing a partition.";
+
+    let parameters = (ins
+        "int":$lweDim,
```

Comment:
We need at least the macro parameters that gives the size of the ciphertext and the noise level (either as manp or as a float). Others will not be used by the optimizer since they could be blocking. E.g. even for PBS parameter of this tfhe.rs partition, concrete can have its own parameters (using the same key as tfhers).